addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairIntelliChlorHandler.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.pentair.handler;
+
+import static org.openhab.binding.pentair.PentairBindingConstants.*;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.pentair.PentairBindingConstants;
+import org.openhab.binding.pentair.internal.PentairPacket;
+import org.openhab.binding.pentair.internal.PentairPacketIntellichlor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PentairIntelliChlorHandler} is responsible for implementation of the Intellichlor Salt generator. It will
+ * process
+ * Intellichlor commands and set the appropriate channel states. There are currently no commands implemented for this
+ * Thing to receive from the framework.
+ *
+ * @author Jeff James - Initial contribution
+ */
+public class PentairIntelliChlorHandler extends PentairBaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(PentairIntelliChlorHandler.class);
+
+    protected PentairPacketIntellichlor pic3cur = new PentairPacketIntellichlor();
+    protected PentairPacketIntellichlor pic4cur = new PentairPacketIntellichlor();
+
+    public PentairIntelliChlorHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing IntelliChlor - Thing ID: {}.", this.getThing().getUID());
+
+        id = 0; // Intellichlor doesn't have ID
+
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Thing {} disposed.", getThing().getUID());
+        updateStatus(ThingStatus.OFFLINE);
```

Comment:
please do not set the status anymore in the dispose - the framework takes care of that.